### ISSUE TITLE
OffSrceen LoadingStateChanged not triggered with LoadPageAsync() Task

### ISSUE THREAD

**Issue by LeMoussel** [cefsharp/CefSharp#1043]
#### Issue Summary
OffscreenBrowser task

#### Technical details
* CefSharp Version: `2272`
* Operating System: Windows
* x86/x64: x86
* OffScreen: 

#### Steps to Reproduce
C# code:  [LeMoussel / CefSharpOSRex](https://gist.github.com/LeMoussel/598dd0a31c9a8369b262)

If I do 
````C#
using (var browser = new ChromiumWebBrowser("http://httpbin.org"))
{
   await LoadPageAsync(browser).ContinueWith(prevTask => prevTask.Wait());
   var task1 = browser.GetSourceAsync();
   task1.Wait();
   string html = task1.Result;
}
````
 `html ` variable contain HTML source code of httpbin.org

But If I do 
````C#
using (var browser = new ChromiumWebBrowser("dummy:"))
{
   await LoadPageAsync(browser, "http://httpbin.org").ContinueWith(prevTask => prevTask.Wait());
   var task1 = browser.GetSourceAsync();
   task1.Wait();
   string html = task1.Result;
}
````
 `html ` variable contain HTML source code of dummy Not HTML source code of  httpbin.org.

**Comment by jankurianski** [comment]
I don't think this has anything to do with async, I believe it is a similar problem to this stackoverflow question (where you construct the browser with an address and then immediately load another address): http://stackoverflow.com/questions/30329350/cefsharp-load-html-page-issue

Can you try this out in the `C/C++` CEF client and see what happens?  i.e. construct the CEF browser with address of `dummy:` and then immediately call `LoadURL()`?  Difficult, I know, but I don't really feel like this is a CefSharp problem, and is instead a timing issue intrinsic to CEF/Chromium.

**Comment by amaitland** [comment]
You have to wait for the underlying `CEF` browser to be fully initialized before you can issue a `LoadUrl()` call. Try using the `BrowserInitialized` event.

If you look at `LoadUrl()` you'll see that it relies on a `CefBrowser` instance.
https://github.com/cefsharp/CefSharp/blob/cef/2272/CefSharp.Core/ManagedCefBrowserAdapter.cpp#L43

The `CefBrowser` instance isn't available until after `CefLifeSpanHandler::OnAfterCreated` has been triggered see
https://github.com/cefsharp/CefSharp/blob/cef/2272/CefSharp.Core/Internals/ClientAdapter.cpp#L98

So basically the call to `LoadUrl()` is a `noop` until after the browser has been initialized.

**Comment by LeMoussel** [comment]
Alex,
I Try using the `BrowserInitialized` event.like this
```C#
        public static bool isBrowserInitialized = false;

            using (var browser = new ChromiumWebBrowser("dummy:"))
            {
                //// Test 1: Not Work
                browser.BrowserInitialized += BrowserInitialized;
                while (isBrowserInitialized == false) { Thread.Sleep(100); }

                await LoadPageAsync(browser, "http://httpbin.org").ContinueWith(prevTask => prevTask.Wait());

                var task1 = browser.GetSourceAsync();
                task1.Wait();
                string html = task1.Result;
            }

        static void BrowserInitialized(object sender, EventArgs e)
        {
            isBrowserInitialized = true;
        }
```
No success, `html variable` contain HTML source code of dummy Not HTML source code of httpbin.org.

I test this 
```C#
            using (var browser = new ChromiumWebBrowser("dummy:"))
            {
                //// Test 2: Not Work
                while (browser.IsBrowserInitialized == false) { Thread.Sleep(100); }

                await LoadPageAsync(browser, "http://httpbin.org").ContinueWith(prevTask => prevTask.Wait());

                var task1 = browser.GetSourceAsync();
                task1.Wait();
                string html = task1.Result;
            }
```
Hélas ! No success, `html variable` contain HTML source code of dummy Not HTML source code of httpbin.org.

And final
```C#
            using (var browser = new ChromiumWebBrowser("dummy:"))
            {
                //// Test 3: Work
               Thread.Sleep(5000); 

                await LoadPageAsync(browser, "http://httpbin.org").ContinueWith(prevTask => prevTask.Wait());

                var task1 = browser.GetSourceAsync();
                task1.Wait();
                string html = task1.Result;
            }
```
Success !, `html variable` contain HTML source code of httpbin.org.

Is there a more elegant way than using.`Thread.Sleep()`?

**Comment by amaitland** [comment]
> Is there a more elegant way than using.Thread.Sleep()?

I'm sure there is. If you put a few breakpoints in the main events you can get a better understanding of the sequence. Having a look through the source of `cefclient` can be a helpful reference.

Probably useful to document somewhere, so if you have the time to write something on the `wiki` that would be great :+1:

**Comment by amaitland** [comment]
You don't actually need to pass in a `URL` to the `OffScreen` constructor, better to leave out than trying to pass in a dummy one.

https://gist.github.com/amaitland/55895ced49420b7d426e

**Comment by LeMoussel** [comment]
Thank you very much Alex, Work fine.

What would you like me to write something on the Wiki? 
*Rem: I am French and I do not write English well, you will correct me.*

I rather make an CefSharp example of using JavaScript code to get informations about response headers and navigation timing ([W3C Permormance Timing](https://dvcs.w3.org/hg/webperf/raw-file/tip/specs/NavigationTiming/Overview.html#performancetiming))

**Comment by amaitland** [comment]
> I rather make an CefSharp example of using JavaScript code to get informations about response headers and navigation timing (W3C Permormance Timing)

Great :+1:
